app/src/main/java/it/niedermann/owncloud/notes/shared/util/NoteUtil.java

```diff
@@ -123,6 +126,29 @@ public static String getLineWithoutMarkdown(@NonNull String content, int lineNum
         return line;
     }
 
+    /**
+     * Strips all lines beginning with a filled checkbox from the Markdown.
+     *
+     * @param content    String
+     * @return newContent String
+     */
+    @NonNull
+    public static String getContentWithoutCheckedItems(@NonNull String content) {
+        final Pattern pattern = Pattern.compile(
+                "^ *[-+*] \\[[xX]\\].*$",
+                Pattern.MULTILINE
+        );
+        final Matcher matcher = pattern.matcher(content);
+
+        // We can't just replace the matcher with an empty string as this would
+        // let a blank line. To be able to remove the blank line as well, we set
+        // a random-ish token that we will then match with the EOL as well and
+        // remove them all together.
+        return matcher
+                .replaceAll("RANDOM-ISH_TEXT_TO_REPLACE")
+                .replaceAll("RANDOM-ISH_TEXT_TO_REPLACE\n?", "");
```

Comment:
I don't have a better idea, but this looks.... odd...

Reply:
I know but I could not find a better way either